src/server/qgsrequesthandler.h

```diff
@@ -96,33 +96,26 @@ class QgsRequestHandler
     //! @note not available in Python bindings
     virtual void setGetCoverageResponse( QByteArray* ba ) = 0;
 
-    virtual void setDefaultHeaders() {}
-
     //! Set an HTTP header
     virtual void setHeader( const QString &name, const QString &value ) = 0;
 
     //! Remove an HTTP header
-    virtual int removeHeader( const QString &name ) = 0;
-
-    //! Delete all HTTP headers
-    virtual void clearHeaders() = 0;
+    virtual void removeHeader( const QString &name ) = 0;
 
     //! Append the bytestream to response body
     virtual void appendBody( const QByteArray &body ) = 0;
 
-    //! Clears the response body
-    virtual void clearBody() = 0;
```

Comment:
As long as you are returning a not-const pointer that can alter the headers `QMap` and if it's available in the bindings, it looks good to me,  a SIP mapped type for `QMap<QString, QString>` should be already available.

I can't recall why I did not write such an accessor to the `QMap` headers when I first designed this API, maybe I did not want to leave direct manipulation of the headers `QMap` dictionary from client code.

If you write such an accessor, there will be no need for any header manipulation methods (setHeader, removeHeader, clearHeaders) since the client code could directly alter the headers dictionary.

Just to be sure we are on the same page: as long as the possibility to create/edit/delete/retrieve individual header entries from client code and plugins is maintained, choose whatever implementation you feel suit best.